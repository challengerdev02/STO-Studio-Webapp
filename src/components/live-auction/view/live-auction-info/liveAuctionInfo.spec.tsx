import { cleanup, render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { LiveAuctionInfo } from './index';

describe('LiveAuctionInfo component', function () {
  afterEach(cleanup);
  let props = {
    imageUrl: '/book-cover.png',
    title: 'Amethyst: Princess of Gemworld',
    auctioneer: 'macauley',
    itemDescription:
      'Amaya, princess of House Amethyst in Gemworld, and her brother love magical pranks. But when one goes much too far, her parents ground the young royal. Amaya, princess of House Amethyst in Gemworld, and her brother love magical pranks.',
  };

  const setup = (extraProps: Record<string, any> = {}) => {
    const utils = render(<LiveAuctionInfo {...props} {...extraProps} />);
    return {
      ...utils,
    };
  };

  beforeEach(() => {
    Object.defineProperty(window, 'matchMedia', {
      writable: true,
      value: jest.fn().mockImplementation((query) => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // deprecated
        removeListener: jest.fn(), // deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    });
  });

  it('should match with snapshot', () => {
    // @ts-ignore
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  it('should test the LiveAuctionInfo component displays ', () => {
    const { getByTestId } = setup();
    expect(getByTestId('live-auction-info')).toBeInTheDocument();
  });

  it('should test the LiveAuctionInfo component has a title displayed ', () => {
    const { getByTestId } = setup();
    expect(getByTestId('bid-title')).toBeInTheDocument();
  });

  it('should test the LiveAuctionInfo component has an auctioneer text displayed ', () => {
    const { getByTestId } = setup();
    expect(getByTestId('bid-auctioneer')).toBeInTheDocument();
  });

  it('should test the LiveAuctionInfo component has a description displayed ', () => {
    const { getByTestId } = setup();
    expect(getByTestId('item-description')).toBeInTheDocument();
  });
});
