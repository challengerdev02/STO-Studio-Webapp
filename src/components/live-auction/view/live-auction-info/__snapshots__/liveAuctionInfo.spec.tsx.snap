// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LiveAuctionInfo component should match with snapshot 1`] = `
<div>
  <div
    class="sc-bdvvtL cvxyAe"
    data-testid="live-auction-info"
  >
    <h2
      class="sc-gsDKAQ etmZfV"
      data-testid="bid-title"
    >
      Amethyst: Princess of Gemworld
    </h2>
    <span
      class="sc-dkPtRN kiHugN"
      data-testid="bid-auctioneer"
    >
      by @
      macauley
    </span>
    <div
      aria-label="Amaya, princess of House Amethyst in Gemworld, and her brother love magical pranks. But when one goes much too far, her parents ground the young royal. Amaya, princess of House Amethyst in Gemworld, and her brother love magical pranks."
      class="ant-typography ant-typography-ellipsis sc-hKwDye ejdsOt"
      data-testid="item-description"
    >
      Amaya, princess of House Amethyst in Gemworld, and her brother love magical pranks. But when one goes much too far, her parents ground the young royal. Amaya, princess of House Amethyst in Gemworld, and her brother love magical pranks.
      <span
        aria-hidden="true"
        style="position: fixed; display: block; left: 0px; top: 0px; z-index: -9999; visibility: hidden; pointer-events: none; word-break: keep-all; white-space: nowrap;"
      >
        lg
      </span>
      <span
        aria-hidden="true"
        style="position: fixed; display: block; left: 0px; top: 0px; z-index: -9999; visibility: hidden; pointer-events: none; width: 0px; white-space: normal; margin: 0px; padding: 0px;"
      >
        <span
          aria-hidden="true"
        >
          ...
        </span>
        <a
          aria-label="Expand"
          class="ant-typography-expand"
        >
          <span
            class="read_more"
          >
            Read More
          </span>
        </a>
      </span>
    </div>
  </div>
</div>
`;
